build(jenkins): try to run tests on android device

sgtcoolguy opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/[TICKET]

Provide a clear PR title prefixed with `[TICKET]`

**Optional Description:**

sgtcoolguy committed:
build(jenkins): try to run tests on adnroid device

build commented:
<!--
  1 failure:  Tests have failed...
  1 warning:  There is no linke...
  2 messages
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>There is no linked JIRA ticket in the PR body. Please include the URL of the relevant JIRA ticket. If you need to, you may file a ticket on <a href="https://jira.appcelerator.org/secure/CreateIssue!default.jspa">JIRA</a></td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-10854/1/artifact/dist/mobilesdk-8.1.0.v20190418105158-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:x: 36 tests have failed
There are 36 tests failing and 462 skipped out of 3788 total tests.</td>
    </tr>
  </tbody>
</table>

### Tests:

| Classname | Name | Time | Error |
| --- | --- | --- | --- |
| android.device.Titanium.App | pause/resume events | 5.005 | timeout of 5000ms exceeded |
| android.device.Titanium.Geolocation | #reverseGeocoder() | 0.007 | expected false to equal true |
| android.device.Titanium.Geolocation | #forwardGeocoder() | 0.259 | expected false to equal true |
| android.device.Titanium.Network.HTTPClient | send on response | 0.077 | done() invoked with non-Error: [object Object] |
| android.device.Titanium.Network.HTTPClient | save response data to temp directory | 60.063 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | POST multipart/form-data containing Ti.Blob | 60.063 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | callbackTestForPOSTMethod | 60.063 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | callbackTestForGETMethod | 60.063 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | setCookieClearCookieWithMultipleHTTPClients | 60.01 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | clearCookieUnaffectedCheck | 60.005 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | sendData | 60.006 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | requestHeaderMethods | 0.048 | done() invoked with non-Error: Received unexpected response: 0 |
| android.device.Titanium.Network.HTTPClient | responseHeadersBug | 60.006 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | emptyPOSTSend | 60.008 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | largeFileWithRedirect | 60.003 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | TIMOB-19042 | 60.002 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | TIMOB-23214 | 60.01 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | TIMOB-23127 | 60.063 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | downloadLargeFile | 60.063 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.HTTPClient | responseXML | 60.02 | timeout of 60000ms exceeded |
| android.device.Titanium.Network.Socket.TCP | #connect(), #write(), #pump() async | 0.007 | done() invoked with non-Error: [object Object] |
| android.device.Titanium.Network.Socket.TCP | #connect() and #write() async | 0.007 | done() invoked with non-Error: [object Object] |
| android.device.Titanium.Network.Socket.TCP | #connect() and receive data | 0.012 | done() invoked with non-Error: [object Object] |
| android.device.Titanium.Network.Socket.TCP | #connect() and send data | 0.013 | done() invoked with non-Error: [object Object] |
| android.device.Titanium.Network.Socket.TCP | #connect() | 0.022 | done() invoked with non-Error: [object Object] |
| android.device.Titanium.UI | #convertUnits() returns Ti.Platform.DisplayCaps.logicalDensityFactory for 1dip to pixels | 0.001 | expected 4 to be approximately 3.5 ±0.001 |
| android.device.Titanium.UI.ImageView | .image (URL-redirect-autorotate) | 10.01 | timeout of 10000ms exceeded |
| android.device.Titanium.UI.Layout | viewWidth | 0.17 | expected 201 to equal 200 |
| android.device.Titanium.UI.Window | .orientationModes LANDSCAPE_LEFT | 0.173 | expected 1 to equal 3 |
| android.device.Titanium.UI.Picker | Selected index persistance | 10.01 | timeout of 10000ms exceeded |
| android.device.Titanium.UI.TabGroup | add Map.View to TabGroup | 10.01 | timeout of 10000ms exceeded |
| android.device.Titanium.UI.View | animate (top %) | 1.791 | expected 0 to be approximately 540 ±10 |
| android.device.Titanium.UI.View | animate (left %) | 1.801 | expected 0 to be approximately 369 ±10 |
| android.device.Titanium.UI.View | animate (left) | 1.823 | expected 100 to equal 150 |
| android.device.Titanium.UI.View | animate (top) | 1.797 | expected 100 to equal 150 |
| android.device.Titanium.UI.WebView | sslerror | 30.002 | timeout of 30000ms exceeded |

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 421dc0f5e842363f75cb70d957b4d9f097edb092
</p>

Commit:
build(jenkins): execute adb commands on all devices

build(jenkins): fix adb alias

build(jenkins): remove alias

build(jenkins): use adb-all script instead

fix(android): validate Processor.getIndex()

fix(android): only transition on lollipop and above

fix(android): guard setZoomLevel()

fix(android): allow AbstractTabGroup to obtain theme colors on older apis

fix(android): allow retry after network error

fix(android): attempt to fix getAllowExitTransitionOverlap() exception

fix(android): formatting

fix(android): validate cpus

Commit:
build(jenkins): only test main branch on devices

Commit:
build(jenkins): cleanup condition

Commit:
build(jenkins): fix main branch definition

Commit:
build(jenkins): temp test on devices

Commit:
build(jenkins): revert force test on devices

garymathews committed:
Merge branch 'master' into android-device-jenkins

garymathews committed:
Merge branch 'master' into android-device-jenkins

garymathews commented on Jenkinsfile:
```diff
@@ -97,20 +109,20 @@ def unitTests(os, nodeVersion, npmVersion, testSuiteBranch) {
 								} else {
 									// gather crash reports/tombstones for Android
-									sh 'adb -e pull /data/tombstones'
+									sh './adb-all.sh pull /data/tombstones'
```
`./adb-all.sh` will run commands on all connected devices and/or running emulators.

garymathews committed:
Merge branch 'master' into android-device-jenkins

garymathews committed:
Merge branch 'master' into android-device-jenkins

garymathews committed:
Merge branch 'master' into android-device-jenkins

garymathews committed:
build(jenkins): fix formatting

sgtcoolguy committed:
Merge branch 'master' into android-device-jenkins

sgtcoolguy commented:
manually merged to master after cleaning up the commit history manually to break apart the build changes from the fixes on android that it uncovered.
